agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracer.java

```diff
@@ -311,23 +297,34 @@ public static Object intercept(
             Method getServerAddress = getMethod(classLoader, "org.apache.hadoop.ipc.RPC", "getServerAddress", Object.class);
             InetSocketAddress inetSocketAddress = (InetSocketAddress) getServerAddress.invoke(o, rpcProxy);
             return executeMethod(zuper, "hdfs://" + inetSocketAddress.getHostString() + ":" + inetSocketAddress.getPort(),
-                    null, dst, FsAction.ADD_BLOCK.name());
+                    null, dst, FsAction.ADD_BLOCK.name(), null);
         }
     }
 
-    private static Object executeMethod(@SuperCall Callable<?> zuper, String uri, String src, String dst, String fsAction) throws Exception {
+    private static Object callDistributedFileSystem(@SuperCall Callable<?> zuper, @This Object o, String src, String dst, String fsAction) throws Exception {
```

Comment:
Any way to test this? This seems very error-prone and could benefit from automated tests

Reply:
done